kani-compiler/src/kani_middle/stubbing/transform.rs

```diff
@@ -119,35 +113,47 @@ fn check_compatibility<'a, 'tcx>(
 const RUSTC_ARG_PREFIX: &str = "kani_stubs=";
 
 /// Serializes the stub mapping into a rustc argument.
-pub fn mk_rustc_arg(stub_mapping: FxHashMap<String, String>) -> String {
+pub fn mk_rustc_arg(stub_mapping: FxHashMap<DefPathHash, DefPathHash>) -> String {
+    // Serialize each `DefPathHash` as a pair of `u64`s, and the whole mapping
+    // as an association list.
+    let mut pairs = Vec::new();
+    for (k, v) in stub_mapping {
+        let kparts = k.0.as_value();
+        let vparts = v.0.as_value();
+        pairs.push((kparts, vparts));
+    }
     // Store our serialized mapping as a fake LLVM argument (safe to do since
     // LLVM will never see them).
-    format!("-Cllvm-args='{RUSTC_ARG_PREFIX}{}'", serde_json::to_string(&stub_mapping).unwrap())
+    format!("-Cllvm-args='{RUSTC_ARG_PREFIX}{}'", serde_json::to_string(&pairs).unwrap())
 }
 
 /// Deserializes the stub mapping from the rustc argument value.
-fn deserialize_mapping(val: &str) -> FxHashMap<String, String> {
-    serde_json::from_str(val).unwrap()
+fn deserialize_mapping(tcx: TyCtxt, val: &str) -> FxHashMap<DefId, DefId> {
+    type Item = (u64, u64);
+    let item_to_def_id = |item: Item| -> DefId {
+        let hash = DefPathHash(Fingerprint::new(item.0, item.1));
+        tcx.def_path_hash_to_def_id(hash, &mut || panic!())
```

Comment:
Maybe a friendly error would be nice here. Something on the terms that this is likely due to a Kani issue, please file a bug report.

Reply:
I checked, and the panic is caught by our panic handler, so the user does receive a message to that effect

Reply:
Thanks!